tools/notification-configuration/notification-creator/NotificationConfigurator.cs

```diff
@@ -149,12 +164,112 @@ public NotificationConfigurator(AzureDevOpsService service, ILogger<Notification
                 if (persistChanges)
                 {
                     result = await service.UpdateTeamForProjectAsync(pipeline.Project.Id.ToString(), result);
+                    if (purpose == TeamPurpose.ParentNotificationTeam)
+                    {
+                        await EnsureScheduledBuildFailSubscriptionExists(pipeline, result, true);
+                    }
                 }
             }
 
+            if (purpose == TeamPurpose.SynchronizedNotificationTeam)
+            {
+                await SyncTeamWithCodeOwnerFile(pipeline, result, gitHubToAADConverter, persistChanges);
+            }
             return result;
         }
 
+        private async Task SyncTeamWithCodeOwnerFile(BuildDefinition pipeline, WebApiTeam team, GitHubToAADConverter gitHubToAADConverter, bool persistChanges)
+        {
+            using (logger.BeginScope("Team Name = {0}", team.Name))
+            {
+                if (pipeline.Process.Type != PipelineYamlProcessType)
+                {
+                    return;
+                }
+
+                // Get contents of CODEOWNERS
+                logger.LogInformation("Fetching CODEOWNERS file");
+                var managementUrl = new Uri(pipeline.Repository.Properties["manageUrl"]);
+                var codeOwnerEntries = await gitHubService.GetCodeownersFile(managementUrl);
+
+                if (codeOwnerEntries == default)
+                {
+                    logger.LogInformation("CODEOWNERS file not found, skipping sync");
+                    return;
+                }
+                var process = pipeline.Process as YamlProcess;
+
+                logger.LogInformation("Searching CODEOWNERS for matching path for {0}", process.YamlFilename);
+        
+                var codeOwnerEntry = CodeOwnersFile.FindOwnersForClosestMatch(codeOwnerEntries, process.YamlFilename);
+                codeOwnerEntry.FilterOutNonUserAliases();
+
+                logger.LogInformation("Matching Contacts Path = {0}, NumContacts = {1}", process.YamlFilename, codeOwnerEntry.Owners.Count);
+
+                // Get set of team members in the CODEOWNERS file
+                var codeownersDescriptors = new List<String>();
+                foreach (var contact in codeOwnerEntry.Owners)
+                {
+                    if (!codeOwnerCache.ContainsKey(contact))
+                    {
+                        codeOwnerCache[contact] = gitHubToAADConverter.GetUserPrincipalNameFromGithub(contact);
+                        if (!string.IsNullOrEmpty(codeOwnerCache[contact]))
+                        {
+                            codeOwnerCache[contact] = await service.GetDescriptorForPrincipal(codeOwnerCache[contact]);
```

Comment:
Is this supposed to be setting the team member cache entry, not the code owner cache?

Reply:
If not, I'm confused how it's supposed to work, since the cache value is both the input and output here.

Reply:
Or if it's intended this way, perhaps this would be better using an intermediate variable for the `GetUserPrincipalNameFromGithub` rather than setting the cache twice with different types of values.

Reply:
Have a intermediate variable.